lib/modules/dosomething/dosomething_kudos/includes/KudosController.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+class KudosController extends EntityAPIController {
+
+  public function __construct() {
+    parent::__construct('kudos');
+  }
+
+  /**
+   * Overrides create() method in EntityAPIController
+   *
+   * @param array $values
+   * @return bool
+   *
+   * @throws Exception
+   */
+  public function create(array $values = []) {
+    $values += ['is_new' => TRUE];
```

Comment:
So the `create()` method for entity, doesn't return an ideal value to help then provide an appropriate response for the API. At least I was having trouble with it. Might be something we can come back to and refactor to re-use the method if a different approach works better :)